.github/workflows/build.yml

```diff
@@ -7,31 +7,19 @@ jobs:
   build:
     runs-on: ubuntu-latest
     steps:
-      - name: Install Ruby
-        uses: actions/setup-ruby@v1
-        with:
-          ruby-version: '2.7'
-
-      - name: Clone
+      - name: Checkout
         uses: actions/checkout@v2
         with:
           fetch-depth: 0
 
-      - name: Cache node modules
-        uses: actions/cache@v2
-        env:
-          cache-name: cache-node-modules
-        with:
-          # npm cache files are stored in `~/.npm` on Linux/macOS
-          path: ~/.npm
-          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
-          restore-keys: |
-            ${{ runner.os }}-build-${{ env.cache-name }}-
-            ${{ runner.os }}-build-
-            ${{ runner.os }}-
+      - name: Build Image
+        run: docker build -t website-portfolio .
```

Comment:
I am a little bit worried about the time it will take to build your docker image. It is not a good practice to rebuild your docker every time you need it. It would be much better to publish the Docker image to some repository or leverage some existing one. I feel this one should be just enough for your project: https://hub.docker.com/layers/michalsimon/static-website/node14/images/sha256-4e8f4300654de965ddf182793c8d1a79db494d92c5de28c68cb1788024a31f25

Then we can run directly docker compose without having 2 different ways of running docker.

Reply:
I see, thank you. Leveraging this Docker image has improved the speed of the CI.